src/sage/combinat/combinat.py

```diff
@@ -2963,28 +2993,88 @@ def bell_polynomial(n: Integer, k: Integer):
         [[], [1], [3], [1]]
         [[], [1], [3, 4], [6], [1]]
 
+    Further checks for :issue:`37727`::
+
+        sage: # needs sage.combinat sage.arith 
+        sage: bell_polynomial(0, 0)
+        1
+        sage: bell_polynomial(0, 0, True)
+        1
+        sage: bell_polynomial(1, 1)
+        x0
+        sage: bell_polynomial(2, 2, True)
+        x0^2
+        sage: bell_polynomial(5)
+        x0^5 + 10*x0^3*x1 + 15*x0*x1^2 + 10*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
+        sage: sum(bell_polynomial(5, k) for k in range(6))
+        x0^5 + 10*x0^3*x1 + 15*x0*x1^2 + 10*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
+        sage: bell_polynomial(5, None, True)
+        x0^5 + 4*x0^3*x1 + 3*x0*x1^2 + 3*x0^2*x2 + 2*x1*x2 + 2*x0*x3 + x4
+        sage: sum(bell_polynomial(5, k, True) for k in range(6))
+        x0^5 + 4*x0^3*x1 + 3*x0*x1^2 + 3*x0^2*x2 + 2*x1*x2 + 2*x0*x3 + x4
+        sage: bell_polynomial(0).parent()
+        Univariate Polynomial Ring in x0 over Integer Ring
+
+    We verify that Bell polynomials equal their power series definition:
+    - if `f(z) = \sum_{n\geq1} x_{n-1} z^n/n!` then
+      `\exp(f) = \sum_{n\geq0} B_n z^n/n!` and
+      `f^k/k! = \sum_{n\geq k} B_{n, k} z^n/n!`
+    - if `g(z) = \sum_{n\geq1} x_{n-1} z^n` then
+      `1/(1-f) = \sum_{n\geq0} \hat B_n z^n` and
+      `f^k = \sum_{n\geq k} \hat B_{n, k} z^n`
+    (see reference).
+        
+        sage: # needs sage.combinat sage.arith 
+        sage: R = PolynomialRing(QQ, 'x', 6)
```

Comment:
I really meant you should do the substitution. Some of the problems with your version is that it is not the definition, and so it will break as soon as you try to use the definition and make $k$ larger. It probably would be better to do the substitution and then check the equality by `all(sub_fps_coeff == bell_poly ...)`. I would also move this into the `EXAMPLES:` block since it is useful to a user to illustrate the identity, not simply to a developer checking consistency.